### PR TITLE
[Gluten-788] Pass hadoop config from driver to executor.

### PR DESCRIPTION
**Pull request by Yohahaha**
## What changes were proposed in this pull request?

Before this pr, we need set extra env variable `VELOX_HDFS` in driver or executor, but that way does not make sense, we should read it from HADOOP_CONFIG eg `core-site.xml` directly.

Spark executor's `SparkConf` is initialized from driver's `SparkConf`, so we could use DriverPlugin to inject configs of what we need to used in executor.

## How was this patch tested?

manual test with hadoop cluster.

I would glad to hear your opinions. :)

### CODE REVIEW

File: cpp/velox/compute/VeloxBackend.cc [hunk 1]
```diff
@@ -80,13 +80,14 @@ void VeloxInitializer::Init(std::unordered_map<std::string, std::string> conf) {
   filesystems::registerHdfsFileSystem();
   // TODO(yuan): should read hdfs client conf from hdfs-client.xml from
   // LIBHDFS3_CONF
-  std::string hdfsUri = "localhost:9000";
+  std::string hdfsUri = conf["spark.fs.defaultFS"];
```
Comment (zhouyuan) [inline cpp/velox/compute/VeloxBackend.cc:83]: does this mean gluten can not read from other HDFS endpoint, only "defaultFS" can work?

Comment (Yohahaha) [reply cpp/velox/compute/VeloxBackend.cc:83]: Thanks your comment.
Gluten could still read from other HDFS endpoint by set env variable `VELOX_HDFS` explicitly, this pr does not change this behavior, it just add a new way to read from `core-site.xml` if we are in fully-prepared hadoop cluster.

btw I'm confused about why use env variable to pass conf, gluten already support pass conf from spark to native backend.

**Comment by Yohahaha** [comment]
Hi @zhouyuan 
I adjusted the order of reading config in newly commit, it would be spark-defaults.conf -> ${HADOOP_CONF_HOME} -> env variables, and env variables have top priority.

**Review by zhouyuan — APPROVED** [review]
👍

**Pull request merged**